### PR TITLE
Feature/tests on bellecour fork

### PR DESCRIPTION
**Pull request by PierreJeanjacquot**
### Changed

- fix `--gas-price` option not being enforced on every command
- provider options for ethers provider are loosely type-checked to allow better control
- Typescript fixes
- move tests to the local fork of bellecour

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -44,4 +33,7 @@ test/tests-working-dir
 dist/**
 IExec*/**
 utils/**
-errors/**
\ No newline at end of file
+errors/**
```
Comment (Le-Caignec) [inline .gitignore:36]: why removing errors ? No file named errors seems to be removed ?

File: package.json [hunk 2]
```diff
@@ -141,15 +141,18 @@
...
-    "generate-json-es-modules": "rimraf src/common/generated && node generateEsModulesFromJson.cjs",
-    "generate-dist": "rimraf dist && mkdir dist && mkdir dist/esm && cp -r src/. dist/esm/",
-    "generate-packages": "node generatePackages.cjs",
-    "generate-doc": "npm run generate-lib-doc && npm run generate-cli-doc",
-    "generate-lib-doc": "./node_modules/typedoc/bin/typedoc --tsconfig tsconfig.doc.json --disableSources",
-    "generate-cli-doc": "node generateCliDoc.cjs"
+    "build": "npm run build:codegen && npm run build:dist && npm run build:packages && npm run build:doc",
+    "test": "npm run test:prepare && NODE_OPTIONS=--experimental-vm-modules jest --coverage",
+    "stop-test-stack": "cd test && docker compose down --remove-orphans",
+    "start-test-stack": "cd test && npm run stop-test-stack && node scripts/prepare-test-env.js && docker compose up -d",
```
Comment (Le-Caignec) [inline package.json:148]: this command will make a `cd test` twice time

File: test/cli/cli-iexec-account.test.js [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+      );
+      expect(initialWalletBalance.add(bnAmount).eq(finalWalletBalance)).toBe(
+        true,
+      );
+      expect(initialAccountBalance.sub(bnAmount).eq(finalAccountBalance)).toBe(
+        true,
+      );
+    });
+
+    test('5 RLC', async () => {
```
Comment (Le-Caignec) [inline test/cli/cli-iexec-account.test.js:167]: it could be cool to test a random value instead

**Review by Le-Caignec — APPROVED** [review]
A little difficult but I noted some remarks on my side

File: test/cli/cli-iexec-account.test.js (hunk 3 above)
Comment (PierreJeanjacquot) [reply test/cli/cli-iexec-account.test.js:167]: the test didn't change, IMO this is out of this PR scope

File: package.json (hunk 2 above)
Comment (PierreJeanjacquot) [reply package.json:148]: Not an issue, the first command `npm run start-test-stack` spawns the second `run stop-test-stack` and they don't share the same working directory.
Every npm script uses the nearest parent `package.json` location as the working directory (in this case the repo root dir).

File: .gitignore (hunk 1 above)
Comment (PierreJeanjacquot) [reply .gitignore:36]: looking at the whole diff this didn't change

**Pull request merged**